rtmp-client/src/main/cpp/flvmuxer/xiecc_rtmp.c

```diff
@@ -665,5 +665,5 @@ int rtmp_sender_write_video_frame(uint8_t *data,
         nal = get_nal(&nal_len, &buf_offset, buf, total);
     }
 
-    return (val > 0) ? 0: -1;
+    return (val > 0) ? RTMP_SUCCESS: RTMP_ERROR_GENERIC;
```

Comment:
Ditto about returning error code returned by `RTMP_Write`. I actually don't understand some of the code above. I think `RTMP_Write` (and possibly `send_key_frame`?) can return negative error results, but e.g. on L611 and L659 it looks like we blindly add the return value into `val`, which might already be non-zero.

Shouldn't these lines be something more like:

```
int ret = RTMP_Write(....);
if (ret is an error code) {
  // handle it somehow
}
val += ret;
...
```

Reply:
Yes, I will make this change.